tensorflow/python/ops/init_ops.py

```diff
@@ -340,3 +341,45 @@ def __call__(self, shape, dtype=dtypes.float32, partition_info=None):
     """Generate a tensor used to initialize a variable."""
     return random_ops._random_walk(shape, self._nonlinearity, dtype,
                                    seed=self._seed)
+
+
+def orthogonal_initializer(gain=1.0, dtype=dtypes.float32, seed=None):
+  """Returns an initializer that generates an orthogonal rotation matrix.
+
+  Initialize a tensor using an orthogonal matrix.
+
+  Args:
+    gain: multiplicative factor to apply to the orthogonal matrix
+    dtype: The type of the output.
+    seed: A Python integer. Used to create random seeds. See
+      [`set_random_seed`](../../api_docs/python/constant_op.md#set_random_seed)
+      for behavior.
+
+  Returns:
+    An initializer that generates orthogonal tensors
+
+  Raises:
+    ValueError: if `dtype` is not a floating point type.
+  """
+  def _initializer(shape, dtype=_assert_float_dtype(dtype), partition_info=None):
+    # Flatten the input shape with the last dimension remaining its original shape so it works for conv2d
```

Comment:
I would be keen to keep the functionality as is but document more clearly as you suggested. The main reason is that I find

```python
weights = tf.get_variable('weights', (5, 5, 8, 32), tf.float32, 
                          tf.orthogonal_initializer())
```

much more straightforward than

```python
weights = tf.reshape(
  tf.get_variable('weights', (200, 32), tf.float32, tf.orthogonal_initializer()), 
  (5, 5, 8, 32)
)
```